### PR TITLE
[api] Fix logit outputs

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch Description**
@tbmihailov pointed out that our logits differ from the openai's, in that we (previously) started our logprobs with the first generated token, rather than including a sentinel `null` logprob and including the first token.

This PR fixes that so we are more consistent with OpenAI

**Testing steps**
So much manual testing.

### CODE REVIEW

File: metaseq/sequence_generator.py [hunk 1]
```diff
@@ -141,27 +141,23 @@ def _generate(
 
         # initialize buffers
         scores = (
-            torch.zeros(bsz * beam_size, max_len + 1).to(src_tokens).float()
+            torch.zeros(bsz * beam_size, max_len).to(src_tokens).float()
         )  # +1 for eos; pad is never chosen for scoring
```
Comment (suchenzang) [inline metaseq/sequence_generator.py:145]: Does that +1 comment for eos still apply?

File: metaseq/sequence_generator.py [hunk 2]
```diff
@@ -254,8 +254,6 @@ def _generate(
                 model_out, log_probs=True, sample=None
             )
             lprobs = lprobs[:, -1, :]
-            if self.need_logprobs:
```
Comment (suchenzang) [inline metaseq/sequence_generator.py:257]: Conditionally seem to be missing the branch of `step >= max_len and self.need_logprobs == True` now - is that intended?

File: metaseq/hub_utils.py [hunk 3]
```diff
@@ -700,23 +711,22 @@ def _filter_special(
...
         output = []
         mask = []
-        for t, s in zip(tokens, scores):
+        for i, (t, s) in enumerate(zip(tokens, scores)):
             if t == pad_token:
                 # simply skip pads
                 mask.append(False)
                 continue
-            if t <= 3:
+            if t <= 3 and i > 0:
```
Comment (suchenzang) [inline metaseq/hub_utils.py:721]: This seems brittle, but probably making it less brittle requires too much glue.  We have asserts to make sure we're using the matching dictionary in streaming lm: https://github.com/facebookresearch/metaseq/blob/d2c75c2ca38c5a68e18c6c4aee2e06e78e2d7ab5/metaseq/tasks/streaming_language_modeling.py#L158-L166 - can we do something similar here?

**Review by suchenzang — APPROVED** [review]

File: metaseq/hub_utils.py (hunk 3 above)
Comment (stephenroller) [reply metaseq/hub_utils.py]: nvm i just did the right thing

File: metaseq/sequence_generator.py (hunk 2 above)
Comment (stephenroller) [reply metaseq/sequence_generator.py:257]: yes. the step == max_len step is forced to be an EOS token regardless. So we don't need to record its score into the logprobs (that's also why scores is protected from writing)

**Comment by dongs0104** [comment]
hello @stephenroller i have question about this,
 when i using huggingface model, is it require to fix or not?
Thanks

**Comment by stephenroller** [comment]
If you're using HF Transformers, then it's their own generation code.
